Create token-explained.md (#77)
solana-foundation/developer-content: content/guides/intro/token-explained.md

---
date: Feb 21, 2024
difficulty: intro
title: "Tokens Explained"
description:
  "Minimal Solana token program skeleton: entry point, parameters for accounts,

>   and instruction data."

Reviewer: This discription is out of date too.

### Analogy: Language Barrier

To better grasp the concept, let's use an analogy. Imagine you speak English
(like a traditional website), and the Solana blockchain speaks its own language
(the Solana programming language). When you want to inquire about your coin
balance, think of the Solana RPC (Remote Procedure Call) as a translator that
bridges the language gap between you and the blockchain.

**The process:**

- **You ask your question:** You click "check balance" on a website. The website
  sends your question in English to the translator (RPC).
- **Translation:** The RPC changes your question into the Solana blockchain's
  language.
- **Asking the blockchain:** The RPC sends the translated question to the Solana
  blockchain.
- **Blockchain's answer:** The blockchain finds your balance and sends the
  answer back to the RPC.
- **Translating back:** The RPC translates the answer back into English.
- **Website gets the info:** The RPC sends the translated answer to the website.
- **You get your answer:** The website shows you your balance in a way you
  understand!

**Key point:** The Solana RPC makes everything work smoothly, so you don't have

> to worry about learning the complicated language of the blockchain.

Reviewer: I would remove this entire section. You're focusing on the different types of accounts here - talking about RPCs isn't necessary.

> ### Analogy: Language Barrier
> 
> To better grasp the concept, let's use an analogy. Imagine you speak English
> (like a traditional website), and the Solana blockchain speaks its own language
> (the Solana programming language). When you want to inquire about your coin
> balance, think of the Solana RPC (Remote Procedure Call) as a translator that
> bridges the language gap between you and the blockchain.
> 
> **The process:**
> 
> - **You ask your question:** You click "check balance" on a website. The website
>   sends your question in English to the translator (RPC).
> - **Translation:** The RPC changes your question into the Solana blockchain's
>   language.
> - **Asking the blockchain:** The RPC sends the translated question to the Solana
>   blockchain.
> - **Blockchain's answer:** The blockchain finds your balance and sends the
>   answer back to the RPC.
> - **Translating back:** The RPC translates the answer back into English.
> - **Website gets the info:** The RPC sends the translated answer to the website.
> - **You get your answer:** The website shows you your balance in a way you
>   understand!
> 
> **Key point:** The Solana RPC makes everything work smoothly, so you don't have
> to worry about learning the complicated language of the blockchain.

Reviewer: ```suggestion
```

## Exploring Different Types of Accounts in Solana

![image](https://github.com/Ayushjhax/developer-content/assets/116433617/cf253051-a943-43d1-a56e-37e8adc39f46)

Now, let's dive into the different types of accounts in Solana, each playing a
crucial role in the ecosystem.

### Wallet Account

> This is like your main wallet where you stash your SOL coins (Solana's own

Reviewer: I can do a full PR review later but this won’t pass legal.

> This is like your main wallet where you stash your SOL coins (Solana's own
> money). It’s used to derive the accounts for your tokens.

Reviewer: ```suggestion
This is your main wallet where you store your SOL (Solana's native token). The address of this account is used to derive the accounts for your tokens.
```

### Token Account

> These accounts are associated with users or wallets and hold vital information
> about the tokens. They work in coherence with the wallet account and mint
> account, sharing similar data and intent for performing transactions.

Reviewer: ```suggestion
These accounts are associated with wallets and hold a balance of a specific token. Their address is calculated from the wallet account and the mint account for that particular token.
```

> ### Mint Account
> 
> The mint account stores data about the token itself, including properties such
> as mint authority, supply, decimals, and freeze authority. Think of these as the
> instruction manuals for each token. They tell you how many tokens are out there,
> who's allowed to make more, that sort of thing.

Reviewer: ```suggestion
### Token Mint Account

The token mint account (sometimes just called a mint account) is like a factory for a specific token. It stores the supply of the token (how many tokens have been created), the mint authority (the account allowed to mint more tokens), the decimals of the token, and (if using Token Extensions Metadata) on-chain metadata about the token, like a name and symbol and URI.
```

Reviewer: Much of the information above is already included in your bullet points.

![image](https://github.com/Ayushjhax/developer-content/assets/116433617/e325ff8a-c020-4ca6-91ad-dda7ea83a32b)

> - **Mint authority:** The designation of your own or some other program using
>   which the mint tokens can be signed.
> - **Supply:** Holds information on how many tokens have been issued overall.
> - **Decimals:** The limit until which the token can be broken down into.
> - **Is Initialized:** Flag to see if the account is ready to go (or not).
> - **Freeze authority:** The designation of your own or some other program using
>   which the mint tokens can be frozen.

Reviewer: ```suggestion
- **Mint authority:** a wallet or multisig account that is allowed to make (mint) more tokens.
- **Supply:** how many tokens have been issued overall
- **Decimals:** The limit until which the token can be broken down into.
- **Freeze authority:** a wallet or multisig account that is allowed to freeze token accounts.
- **Metadata** (if using Token Extensions Metadata) on-chain metadata about the token, like a name and symbol and URI. 
```

> ### Metadata Account
> 
> These accounts hold metadata of mint accounts, storing information about the
> token, such as its name, symbol, description, and image. These are like fancy
> labels on your token boxes. They've got the token's name, its ticker symbol
> (like RAD for RadCoin), maybe even a cool picture.
> 
> **Example:** Currency has a name (US Dollar), symbol ($ or USD), description,
> and image. Solana’s metadata holds another layer of metadata:
> 
> - NAME
> - SYMBOL
> - DESCRIPTION
> - IMAGE

Reviewer: ```suggestion
```

We now suggest using token extensions for token metadata.

> ## Deep Dive into Tokens
> 
> ### Creation and Functionality

Reviewer: ```suggestion
## The Token Program and Token Extensions Program
```

> Tokens in Solana are created using the Token Program, which acts as the Token
> Factory. This program allows users to mint, transfer, and burn tokens between
> accounts, functioning as a factory of accounts that enable token
> functionalities.

Reviewer: ```suggestion
Tokens in Solana are created using the Token Program. This program allows users to create token mints, mint tokens, transfer tokens, and burn tokens.
```

Reviewer: I've removed the 'factory' analogy as we use this elsewhere (not just in this document) for token mint accounts.

![image](https://github.com/Ayushjhax/developer-content/assets/116433617/ece10ac7-028d-4cb1-a737-54cdf6fc6c28)

> #### The Solana Token Machine
> 
> Solana has a special tool called the Token Program that handles all the token
> stuff. Think of it like a factory:
> 
> - **Minting:** This is like printing brand-new tokens based on those instruction
>   manuals (Mint Accounts).
> - **Transferring:** Like moving tokens from your RadCoin box to your friend's.
> - **Burning:** Kind of like tossing tokens in the shredder – they're gone
>   forever!

Reviewer: ```suggestion
 - Create new types of token, by making **token mint accounts**
 - Mints for new tokens (if they're a mint authority for the token mint).  
 - Transfer tokens between accounts 
 -  Burn tokens - destroy tokens forever, and get back the storage fees - often done for spam NFTs! 
```

  ![image](https://github.com/Ayushjhax/developer-content/assets/116433617/012918dc-cc23-4ca9-b6c5-7c9c2052e4aa)

> ### Analogy: Token Factory
> 
> Just as a factory creates products, the Token Program creates accounts for
> tokens in Solana, providing a seamless mechanism for managing tokens within the
> ecosystem.
> 
> ## Solana Token Program
> 
> This Rust program demonstrates the functionality of a token program in the
> Solana blockchain.
> 
> ## Overview
> 
> The program allows minting, transferring, and burning of tokens based on
> provided instruction data. It interacts with Solana accounts using the Solana
> Program library.

Reviewer: ```suggestion
```

Reviewer: This is redundant.
